### PR TITLE
Add NAME type to StringArray codec

### PR DESCRIPTION
**Pull request by davecramer**
@mp911de can you point me in the right direction here. expectNext doesn't expect the right thing?
attempt to fix issue #361 reading array_agg

### CODE REVIEW

**Comment by mp911de** [comment]
I think the latest Docker version on Mac has changed something that requires a version bump to Testcontainers 1.15.

**Comment by davecramer** [comment]
Actually the title of this is a bit mis-leading. I should break this into two PR's. I had to update testcontainers to test this on my mac. The real issue is how to fix the array_arg stuff

File: src/test/java/io/r2dbc/postgresql/ArrayAggIntegrationTests.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        .verifyComplete();
+
+    connection.createStatement("SELECT array_agg(t) FROM my_table")
+        .execute()
+        .flatMap(it -> it.map((row, rowMetadata) -> row.get("array_agg")))
+        .as(StepVerifier::create)
+        .expectNext(new Object[]{"hello", "world", null})
+        .verifyComplete();
+
+    connection.close().block();
```
Comment (davecramer) [inline src/test/java/io/r2dbc/postgresql/ArrayAggIntegrationTests.java:50]: @mp911de I'm not sure how to do this bit?

File: src/test/java/io/r2dbc/postgresql/ArrayAggIntegrationTests.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import io.r2dbc.postgresql.api.PostgresqlConnection;
+import io.r2dbc.postgresql.api.PostgresqlResult;
+import io.r2dbc.postgresql.codec.Json;
+import io.r2dbc.postgresql.util.PostgresqlServerExtension;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.springframework.jdbc.core.JdbcOperations;
+import reactor.test.StepVerifier;
+
+public class ArrayAggIntegrationTests {
```
Comment (mp911de) [inline src/test/java/io/r2dbc/postgresql/ArrayAggIntegrationTests.java:12]: You can find the formatter config at `intellij-style.xml`

File: src/test/java/io/r2dbc/postgresql/ArrayAggIntegrationTests.java (hunk 1 above)
Comment (mp911de) [reply src/test/java/io/r2dbc/postgresql/ArrayAggIntegrationTests.java:50]: You can subclass `AbstractIntegrationTests`. That will give you access to a `connection` field so you don't need to worry about the connection lifecycle. Other than that, `close.block()` (or `close().as(StepVerifier::create).verifyComplete()`) are fine.

**Comment by mp911de** [comment]
Do you want to split the PR or should I take care of the dependency upgrade and you simply rebase your PR?

**Comment by davecramer** [comment]
Do the latter. Update the dependency. It's irrelevant to my PR. Thanks

**Comment by mp911de** [comment]
Upgraded to Testcontainer 1.15.0 via #363.

**Comment by davecramer** [comment]
Ok so how does one do expectNext( some array ) ? The value returned is an array of strings with 3 elements. Alternatively how does one do expectNext(null) ?

**Comment by mp911de** [comment]
`expectNext` does an `equals` check, so probably not ideal. Use `consumeNextWith` or `assertNext` as in:

```
            .assertNext(json -> {

                assertThat(json.data).contains("{ \"customer\": \"John Doe\"");

            })
```


Reactive Streams prohibits the usage of `null`. Instead of emitting null values, rather do not emit the element (or wrap it in `Optional`).

File: src/test/java/io/r2dbc/postgresql/ArrayAggIntegrationTests.java (hunk 1 above)
Comment (davecramer) [reply src/test/java/io/r2dbc/postgresql/ArrayAggIntegrationTests.java:50]: FYI, I copied `JsonIntegrationTests` which apparently don't subclass `AbstractIntegrationTests`

Comment (mp911de) [reply src/test/java/io/r2dbc/postgresql/ArrayAggIntegrationTests.java:50]: I filed #364 to fix `JsonIntegrationTests`. I think, previously we had no config hook to force binary transfer.

**Comment by davecramer** [comment]
So that works but it doesn't expose the bug. Thoughts ?

**Comment by mp911de** [comment]
Adding the `NAME_ARRAY` seems about right. I'd go forward and merge this one. During the merge I'd also update the exception message to report the oid if a value cannot be decoded to aid debugging.

**Comment by davecramer** [comment]
I'm not particularly happy about the name of the new  test. I'll change that and add the oid in the exception and merge it.

**Comment by mp911de** [comment]
Great, thanks for letting me know. For issues and pull requests we use `[#???]` and `[resolves #???]` syntax to associate commit messages to tickets/pull requests.

**Comment by davecramer** [comment]
What do you mean by cherry-picking ? You don't use the GUI to merge them ?

**Comment by mp911de** [comment]
Exactly. I found that the easiest way is to `git cherry-pick <sha>` to bring commits into branches. This PR requires also backporting into `0.8.x` once it's merged.

**Pull request closed without merging**